File: 151004/Terebey/JWT/publisher/src/main/java/org/education/bean/dto/MessageRequestTo.java
package org.education.bean.dto;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.Size;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class MessageRequestTo {
    int id;

    @Min(value = 0)
    int issueId;

    @Size(min = 2, max = 2048, message = "Incorrect content size")
    String content;
}

File: 151004/Burbouski/appLab3/publisher/src/main/kotlin/com/github/hummel/dc/lab3/dto/request/MessageRequestTo.kt
package com.github.hummel.dc.lab3.dto.request

import com.github.hummel.dc.lab3.bean.Message
import kotlinx.serialization.Serializable

@Serializable
data class MessageRequestTo(
	private val issueId: Long, private val country: String, private val content: String
) {
	fun toBean(id: Long?): Message = Message(
		id, country, issueId, content
	)

	init {
		require(content.length in 4..2048)
	}
}